### PR TITLE
Added front page to site

### LINKED ISSUE #35: Create front page

### PR DESCRIPTION
**Pull request by aalvarado**
This displays a specially crafted page that can be displayed as the main page. It is different from all of the pages. closes #35 #34

### CODE REVIEW

File: app/models/page.rb [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
+
+  def self.main_page
+    where( :main_page => true ).first
+  end
+
+  def check_main_page
+    if main_page?
+      page = Page.where("main_page = ?", true).first
+      page.update_attributes(:main_page => false) if page.present? && page != self
+    end
```
Comment (EnriqueVidal) [inline app/models/page.rb]: Even though it doesn't look too pretty, this is possible a better way to do it:

``` ruby
def check_main_page
  Page.where( :main_page => true ).find_each( :batch_size => 100 ) do |page|
    page.update_attributes :main_page => false
  end

  self.main_page = true
end
```

I **DON'T** like to not assume that  only one `main_page` exists.

**Comment by EnriqueVidal** [comment]
Needs test, and needs to fix existing broken test when you changed the `root_path`

File: app/models/page.rb (hunk 1 above)
Comment (EnriqueVidal) [inline app/models/page.rb]: Actually I take that back I think your original idea was fine, since I asked you to change this I'm rolling it back.

**Comment by EnriqueVidal** [comment]
The build passes[1656654](http://travis-ci.org/#!/rubytij/usergroup/jobs/1656654)

**Comment by EnriqueVidal** [comment]
I only wrote enough to make it pass, here are my observations:
1. We need to rename the timestamps in the events from `start_date` and `ende_date` to `starts_at` and `ends_at` the reason for that is that they're time not dates.
2. We have a relationship with user (as in owner I think), but it doesn't mean that whoever created will give the talk.
3. We should have an approved boolean and not just accept anything that comes in through the website.

**Comment by EnriqueVidal** [comment]
@aalvarado `app/assets/fonts/bebas-neue/demo.html` and `app/assets/fonts/bebas-neue/Dharma Type Font License.txt`?

Also we should put this under `vendor/assets/fonts`.

**Comment by EnriqueVidal** [comment]
we should actually move the whole `bebas-neue` folder to `vendor`.

**Comment by aalvarado** [comment]
@EnriqueVidal ok, I'll move it to vendor.

**Comment by EnriqueVidal** [comment]
The navigation and the github logo are all relative paths which means they work in some pages and fail in others.

**Comment by EnriqueVidal** [comment]
We need to revisit the root_path thing, we still have the `main_page` field in the table, but we can't make any other page the main_page because it's tied to a page name and section name, what I don't like about this approach is that  if the page name changes or the page is remove we will end up with a broken app, there if we ever want a different page to be the main_page we would need to remove the existing one and create a new one with the name of the old page, or... edit the routes and add the new one and neither solution strike me as the way to go, @aalvarado can we reset the routes to the way they were when we had a `main_page` method? I already took care of some of the stuff that were pending we only really need this and to limit the acces on the `dashboard/pages` controller with `acl9` to only `owners` and `editors`.

**Comment by EnriqueVidal** [comment]
Also instead of have `.photos` div with a background image hardcoded in the stylesheet we should have one image per section named after the section and render it regardless wether it is the `main_page` or not, I'm talking about having something like `about.png`, `home.png`, `contact.png` etc. that will keep layout consistent and will prevent us from using `if`'s in our layouts.

**Comment by EnriqueVidal** [comment]
We need flash messages too. right now they're being ignored.

**Comment by aalvarado** [comment]
@EnriqueVidal Yes, I agree on the `main_page` method. do you want me to take a look at the access control for owners on pages? 

Regarding the `.photos` div, I agree that we could have something there for different pages, but I would like to see a smaller banner size, I dont think there's a need for a same height banner on every page, we would be wasting space for content.

We can create an issue for that; we don't need it for this PR. And I ( or anyone else interested ) will take care of it after this PR is merged.

**Comment by EnriqueVidal** [comment]
@aalvarado agreed, lets focus on returning `main_page` to work as originally intended, the `access_control` call is really simple:

``` ruby
access_control do
  allow :owners, :editors
end
```

In the `PagesController`, but you'll need to update our tests to take this into account, I can do that if you like.

**Comment by aalvarado** [comment]
I can do it, but I think I can work on this until friday.

**Comment by EnriqueVidal** [comment]
Test coverage seems good and the build is passing, I will merge this and open another issue for some CSS improvements and locales and layout cleanup.

**Pull request merged**